Api/Resilience.md

```diff
@@ -0,0 +1,117 @@
+
+We can install Polly through the NuGet package manager or dotnet CLI:
+
+```bash
+Install-Package Polly
+```
+
+or
+
+```bash
+dotnet add package Polly
+```
+
+### Retry Mechanism
+
+The Retry policy is one of the most important aspects of any resilient system. It allows us to retry a failed request a specified number of times and wait for some time before executing another request. For example, we are trying to execute an HTTP request to an external service that is down at the moment.
+
+In this case, we want to retry our operation with Polly. Initially, create the `AsyncRetryPolicy` to specify configuration and behavior:
+
+``` c#
+	AsyncRetryPolicy retryOnExceptionPolicy =
+		Policy.Handle<Exception>().RetryAsync(5);
+```
+
+A configuration like this will perform the retry in the event of an `Exception`, up to five times.
+
+With the `AsyncRetryPolicy` configured, we can wrap our external service call:
+
+``` c#
+	await retryOnExceptionPolicy
+		.ExecuteAsync(async () => await _customerService.GetAll());
+```
+
+Our external API call is now much more resilient. In addition to that, Polly offers some advanced retry functionalities:
+
+- Retry forever (until the operation is successful)
+- Wait and retry - waits for a specified duration between retries
+  
+What happens if the error is authentication related for example? It doesn't make sense to retry an action that won't succeed. However, we can use the `onRetry` delegate from Polly:
+
+``` c#
+	AsyncRetryPolicy retryOnExceptionPolicy =
+		Policy.Handle<Exception>().RetryAsync(5, onRetry: (response, retryCount) => 
+		{
+			if (response.Result.StatusCode == HttpStatusCode.Unauthorized) 
+			{
+				// Execute re-authentication
+			}
+		});
+```
+
+The retry policy is not always useful and should be avoided in some cases. For example, it doesn't make sense to keep retrying when the external service produces an error that stands for a long time. That said, the retry policy is not a long-term solution for issues but can be used as a mitigation for accidental error occurrences.
+
+### Circuit Breaker pattern
+
+Circuit Breaker is another very important resiliency pattern. It comes in handy when the external API is unreliable and it's better to stop spamming the service for a specified amount of time. If the service is under heavy load, we should stop for a while, and try again later. Each house is powered by electricity nowadays and has a circuit breaker. The main task of the circuit breaker is to stop everything when an issue occurs to protect the electrical devices. The Circuit Breaker pattern works in the same way.
```

Comment:
Hey, that makes sense. I restructured the article and removed some parts in order to focus more on the core approaches, rather than Polly. Polly is still used to provide examples from code, though.